libcontainer/cgroups/systemd/unified_hierarchy.go

```diff
@@ -151,6 +151,9 @@ func (m *UnifiedManager) Apply(pid int) error {
 	if err := createCgroupsv2Path(path); err != nil {
 		return err
 	}
+	if err := cgroups.WriteCgroupProc(path, pid); err != nil {
```

Comment:
isn't it done automatically by systemd?

I see we are specifying the PID here: https://github.com/opencontainers/runc/blob/master/libcontainer/cgroups/systemd/unified_hierarchy.go#L61-L63

Reply:
I see that systemd v1 driver (aka `LegacyManager`) is doing it explicitly:
* Apply -> joinCgroups -> join for all controllers except cpuset
* ... -> cpuset.ApplyDir for cpuset

https://github.com/opencontainers/runc/blob/ccbb3364d49d2dc6d9f057134570b0f382f6ceb7/libcontainer/cgroups/systemd/apply_systemd.go#L312

Reply:
I guess it is done because some controllers are not supported by systemd (like cpuset).  On cgroup v2 it is enough to join the cgroup once.

Reply:
@giuseppe you're right, adding PIDs property should be enough, and yet it is not working. Investigating.

Reply:
I'd investigate if `dbusConnection.StartTransientUnit` waits correctly that the systemd unit is created.  Does it eventually join the correct cgroup if you add a sleep here?

Reply:
See https://github.com/opencontainers/runc/issues/2310#issuecomment-613128742